src/searchers/beam_search.cpp

```diff
@@ -392,7 +396,11 @@ void BeamSearch::searchTopK(std::tuple<float *, int, int> &result) {
         std::vector<long unsigned int> recvCount(msgerSize, static_cast<long unsigned int>(batchSize * numBeams));
         messenger.allgatherv(maxVal, batchSize * numBeams, recvMax, recvCount);
 
-        float sumVal[batchSize * numBeams] = {0};
+        float sumVal[batchSize * numBeams];
+        for (int i = 0; i < batchSize * numBeams; ++i) {
+            sumVal[i] = 0.0f;
```

Comment:
Original float "sumVal[batchSize * numBeams] = {0};" should work.

Reply:
Yes, original code could work, but this is c++20 standard. If we want to build w/ ICX, it should be c++17 standard.